File: book_kit/src/main/java/net/lzbook/kit/user/bean/QGBook.java
package net.lzbook.kit.user.bean;

/**
 * 类描述：
 * 创建人：Zach
 * 智能书籍建造类
 */
public class QGBook extends Book {

    private QGBook(QGBook.Builder builder) {
        book_id = builder.bookId;
        book_source_id = builder.bookSourceId;
        name = builder.bookName;
        category = builder.category;
        author = builder.author;
        chapter_count = builder.chapterCount;
        last_chapter_name = builder.lastChapterName;
        img_url = builder.imgUrl;
        status = builder.status;
        site = builder.host;
        last_updatetime_native = builder.updateTime;
    }

    public static class Builder {
        //最新更新时间
        public long updateTime;
        private String bookId;
        private String bookSourceId;
        private String bookName;
        private String category;
        private String author;
        private int chapterCount;
        private String lastChapterName;
        private String imgUrl;
        private int status = -1;
        private String host;

        public QGBook build() {
            return new QGBook(this);
        }

        public QGBook.Builder bookId(String bookId) {
            this.bookId = bookId;
            return this;
        }

        public QGBook.Builder bookSourceId(String bookSourceId) {
            this.bookSourceId = bookSourceId;
            return this;
        }

        public QGBook.Builder bookName(String bookName) {
            this.bookName = bookName;
            return this;
        }

        public QGBook.Builder category(String category) {
            this.category = category;
            return this;
        }

        public QGBook.Builder author(String author) {
            this.author = author;
            return this;
        }

        public QGBook.Builder chapterCount(int chapterCount) {
            this.chapterCount = chapterCount;
            return this;
        }

        public QGBook.Builder lastChapterName(String lastChapterName) {
            this.lastChapterName = lastChapterName;
            return this;
        }

        public QGBook.Builder host(String host) {
            this.host = host;
            return this;
        }

        public QGBook.Builder imgUrl(String imgUrl) {
            this.imgUrl = imgUrl;
            return this;
        }

        public QGBook.Builder status(int status) {
            this.status = status;
            return this;
        }

        public QGBook.Builder updateTime(long updateTime) {
            this.updateTime = updateTime;
            return this;
        }
    }

}

File: basic/src/main/java/com/ding/basic/bean/QGBook.kt
package com.ding.basic.bean

import com.ding.basic.database.migration.StatusConverter

/**
 * 类描述：
 * 创建人：Zach
 * 智能书籍建造类
 */
class QGBook private constructor(builder: QGBook.Builder) : Book() {

    init {
        book_id = builder.bookId!!
        book_source_id = builder.bookSourceId!!
        name = builder.bookName
        label = builder.category
        author = builder.author
        chapter_count = builder.chapterCount
        img_url = builder.imgUrl
        status = StatusConverter().convert(builder.status)
        host = builder.host
        last_chapter?.update_time = builder.updateTime
    }

    class Builder {
        var bookId: String? = null
        var bookSourceId: String? = null
        var bookName: String? = null
        var category: String? = null
        var author: String? = null
        var chapterCount: Int = 0
        var lastChapterName: String? = null
        var imgUrl: String? = null
        var status = -1
        var host: String? = null
        //最新更新时间
        var updateTime: Long = 0

        fun build(): QGBook {
            return QGBook(this)
        }

        fun bookId(bookId: String): QGBook.Builder {
            this.bookId = bookId
            return this
        }

        fun bookSourceId(bookSourceId: String): QGBook.Builder {
            this.bookSourceId = bookSourceId
            return this
        }

        fun bookName(bookName: String): QGBook.Builder {
            this.bookName = bookName
            return this
        }

        fun category(category: String): QGBook.Builder {
            this.category = category
            return this
        }

        fun author(author: String): QGBook.Builder {
            this.author = author
            return this
        }

        fun chapterCount(chapterCount: Int): QGBook.Builder {
            this.chapterCount = chapterCount
            return this
        }

        fun lastChapterName(lastChapterName: String): QGBook.Builder {
            this.lastChapterName = lastChapterName
            return this
        }

        fun host(host: String?): QGBook.Builder {
            this.host = host
            return this
        }

        fun imgUrl(imgUrl: String?): QGBook.Builder {
            this.imgUrl = imgUrl
            return this
        }

        fun status(status: Int): QGBook.Builder {
            this.status = status
            return this
        }

        fun updateTime(updateTime: Long): QGBook.Builder {
            this.updateTime = updateTime
            return this
        }
    }

}